demos/guide/tls-client-block.c

```diff
@@ -0,0 +1,267 @@
+/*
+ *  Copyright 2023 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ *  Licensed under the Apache License 2.0 (the "License").  You may not use
+ *  this file except in compliance with the License.  You can obtain a copy
+ *  in the file LICENSE in the source distribution or at
+ *  https://www.openssl.org/source/license.html
+ */
+
+/*
+ * NB: Changes to this file should also be reflected in
+ * doc/man7/ossl-guide-tls-client-block.pod
+ */
+
+#include <string.h>
+
+/* Include the appropriate header file for SOCK_STREAM */
+#ifdef _WIN32 /* Windows */
+# include <winsock2.h>
+#else /* Linux/Unix */
+# include <sys/socket.h>
+#endif
+
+#include <openssl/bio.h>
+#include <openssl/ssl.h>
+#include <openssl/err.h>
+
+/* Helper function to create a BIO connected to the server */
+static BIO *create_socket_bio(const char *hostname, const char *port)
+{
+    int sock = -1;
```

Comment:
On Windows the `socket` type is a HANDLE, not an `int`. Is this code supposed to be Windows-portable?

Reply:
Yes, that is an interesting anomaly. However `BIO_socket()` and friends return an `int` - so this code is correct. Internally`BIO_socket()` implicitly casts the `HANDLE` to `int`. But its been that way for a very long time and is used (for example in s_client and s_server). So in practice I don't think this is actually a problem. Either way it isn't a problem for this demo code because it is using the API as it is designed.